### PR TITLE
Add team-related functionality to github module/state

### PR DESCRIPTION
**Pull request by garrettheel**
### What does this PR do?

Adds `team_present` and `team_absent` states along with related module implementations.
### What issues does this PR fix or reference?

N/A
### New Behavior

Allows for managing teams in the github state as well as optionally enforcing MFA requirements on team members.
### Tests written?

Not yet. Open to feedback on where they'd be most useful.

### CODE REVIEW

File: salt/modules/github.py [hunk 1]
```diff
@@ -136,13 +136,16 @@ def _get_repos(profile, params=None):
...
+def list_users(profile="github", ignore_cache=False):
     '''
     List all users within the organization.
 
     profile
         The name of the profile configuration to use. Defaults to ``github``.
 
+    ignore_cache
+        Bypasses the use of cached users.
+
```
Comment (rallytime) [inline salt/modules/github.py]: Can you add a `.. versionadded:: Carbon` tag to this new option?

File: salt/modules/github.py [hunk 2]
```diff
@@ -842,6 +841,537 @@ def list_public_repos(profile='github'):
     return repos
 
 
+def get_team(name, profile="github"):
+    '''
+    Returns the team details if a team with the given name exists, or None
+    otherwise.
+
```
Comment (rallytime) [inline salt/modules/github.py]: Same comment for these new functions: `.. versionadded:: Carbon`. :)

File: salt/states/github.py [hunk 3]
```diff
@@ -144,3 +155,316 @@ def absent(name, profile="github", **kwargs):
...
+        repo_names=None,
+        privacy='secret',
+        permission='pull',
+        members=None,
+        enforce_mfa=False,
+        no_mfa_grace_seconds=0,
+        profile="github",
+        **kwargs):
+    '''
+    Ensure a team is present
```
Comment (rallytime) [inline salt/states/github.py]: Same versionadded comment to these, too.

**Comment by rallytime** [comment]
@garrettheel Awesome! Thank you very much for these additions! I only had some small comments about some doc additions. Otherwise, this looks great!

**Comment by garrettheel** [comment]
@rallytime No problem! Versions have been added

**Comment by rallytime** [comment]
Awesome! We'll get this in soon :)

File: salt/modules/github.py [hunk 4]
```diff
@@ -842,6 +843,561 @@ def list_public_repos(profile='github'):
...
+        organization._requester.requestJsonAndCheck(
+            'POST',
+            organization.url + '/teams',
+            input=parameters
+        )
+        list_teams(ignore_cache=True)  # Refresh cache
+        return True
+    except github.GithubException as e:
+        logging.exception('Error creating a team: {0}'.format(str(e)))
```
Comment (s0undt3ch) [inline salt/modules/github.py]: `log.<level_method>` not `logging.<level_method>`

File: salt/modules/github.py [hunk 5]
```diff
@@ -842,6 +843,561 @@ def list_public_repos(profile='github'):
...
+
+    .. code-block:: bash
+
+        salt myminion github.edit_team 'team_name' description='Team description'
+
+    .. versionadded:: Carbon
+    '''
+    team = get_team(name, profile=profile)
+    if not team:
+        logging.error('Team {0} does not exist'.format(name))
```
Comment (s0undt3ch) [inline salt/modules/github.py]: `log.<level_method>` not `logging.<level_method>`

File: salt/modules/github.py [hunk 6]
```diff
@@ -842,6 +843,561 @@ def list_public_repos(profile='github'):
...
+            parameters['permission'] = permission
+
+        team._requester.requestJsonAndCheck(
+            "PATCH",
+            team.url,
+            input=parameters
+        )
+        return True
+    except UnknownObjectException as e:
+        logging.exception('Resource not found: {0}'.format(team['id']))
```
Comment (s0undt3ch) [inline salt/modules/github.py]: `log.<level_method>` not `logging.<level_method>`

**Comment by s0undt3ch** [comment]
Instead of commenting all occurrences, do `log.<level_method>` instead of `logging.<level_method>` throughout the code, we don't want to log to the root logger.

**Comment by garrettheel** [comment]
@s0undt3ch good call, I fixed the existing and new occurrences of that.

**Comment by cachedout** [comment]
Really nice work here, @garrettheel

**Pull request merged**